tensorflow/compiler/xla/service/pattern_matcher.h

```diff
@@ -2157,6 +2225,22 @@ XLA_NULLOP_PATTERN(PartitionId)
 XLA_NULLOP_PATTERN(ReplicaId)
 #undef XLA_NULLOP_PATTERN
 
+// Matches any of a set of unary ops including the operand and optionally
+// returns the index of the matched op.
+template <typename Arg>
+inline auto UnaryOpAnyOfOrOperand(Arg&& arg, absl::Span<const HloOpcode> ops,
```

Comment:
I'm not sure about the function name; it doesn't make a ton of sense to me.  WDYT about `OptionalUnaryOp`?

Reply:
I think it's a bit vague. I abbreviated the name to `UnaryOpAnyOf`.